RULES_DESCRIPTIONS.md

```diff
@@ -0,0 +1,445 @@
+
+_Description_: Methods or fields of `struct`s that have similar names may be misleading.
+Warns on methods and fields with names that differ only by capitalization.
+
+_Configuration_: N/A
+
+## confusing-results
+
+_Description_: Function or methods that return multiple, no named, values of the same type may induce to error. This rule warns on those cases.
+
+_Configuration_: N/A
+
+## constant-logical-expr
+
+_Description_: Using Boolean literals (`true`, `false`) in logic expressions may make the code less readable. This rule
+suggests removing Boolean literals from logic expressions.
+The rule will also try to spot logical expressions that evaluate always to the same value.
+
+_Configuration_: N/A
+
+## context-as-argument
+
+_Description_: Is a language [convention](https://github.com/golang/go/wiki/CodeReviewComments#contexts) that `context.Context` should be the first parameter of a function. This rule spots function declarations that do not follow the convention.
+
+_Configuration_: N/A
+
+## context-keys-type
+
+_Description_: Basic types should not be used as key in `context.WithValue`.
+This rule warns when that is the case.
+
+_Configuration_: N/A
+
+## cyclomatic
+
+_Description_: [Cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) is a measure of code complexity.
+Enforcing a maximum complexity per function helps keeping code readable and maintainable.
+
+_Configuration_: (int) the maximum function complexity
+
+Example:
+```toml
+[cyclomatic]
+  arguments =[3]
+```
+
+## deep-exit
+
+_Description_: Packages exposing functions that can stop program execution by exiting are hard to reuse. This rule looks for program exits in functions other than `main()` or `init()`.
+
+_Configuration_: N/A
+
+## dot-imports
+ 
+ _Description_: Importing with `.` makes the programs much harder to understand because it is unclear where names belongs to the current package or to an imported package.
+ This rule warns when using imports `.`
+
+ More information [here](https://github.com/golang/go/wiki/CodeReviewComments#import-dot)
+
+_Configuration_: N/A
+
+## empty-block
+
+_Description_: Empty blocks make code less readable and may be the symptom of a bug (unfinished refactoring?). This rule spots empty blocks in the code.
+
+_Configuration_: N/A
+
+## empty-lines
+
+_Description_: Sometimes `gofmt` is not enough to enforce a common formating of a code-base, this rule warns when there are heading or trailing newlines in code blocks.
+
+_Configuration_: N/A
+
+## error-naming
+
+_Description_: By convention, for the sake of readability, variables holding errors must be named with the prefix `err`. This rule warns when this convention is not followed.
+
+_Configuration_: N/A
+
+## error-return
+
+_Description_: By convention, for the sake of readability, when a function returns a value of type `error`, the value must be the last one of the list of return values. This rule warns when this convention is not followed.
```

Comment:
Maybe we can refactor this a bit?

> By convention, for the sake of readability, the errors should be last in the list of returned values by a function.

Or something like this. There's duplication of the word "value" otherwise.